raft_test.go

```diff
@@ -2639,6 +2639,66 @@ func TestRaft_VoteNotGranted_WhenNodeNotInCluster(t *testing.T) {
 	}
 }
 
+// TestRaft_FollowerRemovalNoElection ensures that a leader election is not
+// started when a standby is shut down and restarted.
+func TestRaft_FollowerRemovalNoElection(t *testing.T) {
+	// Make a cluster
+	inmemConf := inmemConfig(t)
+	inmemConf.HeartbeatTimeout = 500 * time.Millisecond
```

Comment:
If you reduce these to maybe 100ms each then you could reduce the 3s wait later on and ensure that this test runs much more quickly.

The total time to get to the capped backoff would be no more than 170ms:
* <10ms - standard wait between heartbeats.
* 10ms - first error wait 
* <10ms - standard wait
*  20ms - second error wait
* <10ms - standard wait
*  40ms - third error wait
* <10ms - standard wait
*  50ms - fourth error wait capped due to 50% of 100ms

Then we repeat the same <60ms wait for each subsequent error. In total there is about 160ms of exponential backoff in there so you could sleep 250ms just to be on the safe side. That would reduce this tests runtime by about 83% if my calculations are correct.

Reply:
Sounds good -- removed the time.Sleep after the node spun down, and changed the time.Sleep to 250 ms during which we're waiting for a re-election.